pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -137,3 +135,61 @@ func (r *DNSPolicyReconciler) deleteGatewayDNSRecords(ctx context.Context, gatew
 	}
 	return nil
 }
+
+func (r *DNSPolicyReconciler) buildClusterGateway(ctx context.Context, upstreamGateway *gatewayv1beta1.Gateway, downstreamCluster string) (dns.ClusterGateway, error) {
+	var target dns.ClusterGateway
+
+	mc := &clusterv1.ManagedCluster{}
+	if err := r.Client().Get(ctx, client.ObjectKey{Name: downstreamCluster}, mc, &client.GetOptions{}); err != nil {
+		return target, err
+	}
```

Comment:
Leaving this here still ties us to MGC. I think the logic around this should be that if we have a multi cluster gateway (You will know this after processing the addresses), we don't check for the existence of a `clusterv1.ManagedCluster` but instead use the gateway itself as the cluster i.e. `dns.NewClusterGateway(upstreamGateway, gatewayAddresses)`.

Admittedly the naming is confused, but `cluster` can be any `metav1.Object` that can have labels attached to it. In the case of a single cluster Gateway, using the gateway itself for this makes the most sense.

Reply:
Would it be a good idea to address it during the #496 ?

Reply:
Created specific follow on for this https://github.com/Kuadrant/multicluster-gateway-controller/issues/622